ocaml/testsuite/tests/typing-layouts/error_message_attr.ml

```diff
@@ -20,7 +58,8 @@ Error: This expression has type float# but an expression was expected of type
        The layout of float# is float64, because
          it is the primitive float64 type float#.
        But the layout of float# must be a sublayout of value, because
-         the [@error_message] attribute: Custom message.
+         of the annotation on the wildcard _ at line 1, characters 20-31
+         Custom message.
```

Comment:
Could we get the `.` to appear on the previous line? I see how that might be hard. If it is hard, then it's OK to keep it here, but also include it on the previous line (which should not be hard).

Why I want to get rid of the final dot: maybe the custom message is actually a question like "Did you mean to use match%optional_u?"

Reply:
By having a special match expression for `With_error_message` in `format_flattened_history`, we can have the message after the `.`